tests/test_core.py

```diff
@@ -11,3 +13,40 @@ def test_grid():
     assert grid.grid_data == grid_conf
     assert grid.r_min == 0.0
     assert grid.r_max == 0.1
+
+
+def test_clock():
+    """Tests for initialization of SimulationClock"""
+    clock_config = {'start_time': 0.0,
+                    'end_time': 1e-8,
+                    'dt': 1e-2,
+                    'print_time': True}
+    with pytest.raises(RuntimeError):
+        SimulationClock(Simulation({}), clock_config)
```

Comment:
It's not obvious what this test is checking for. Is there a why to write this to make it clear _why_ you expect a RuntimeError here?

Reply:
@arichar6 your comment prompted me to think about how we are naming tests. See #70 .

Reply:
@arichar6 @padamson would something like `test_integer_num_steps` be more informative, or is it too lengthy?

Reply:
I'm ok with long test function names, as long as they are informative. You suggestions seems good.

Also, maybe you can do something like:
```suggestion
    clock_config = {'start_time': 0.0,
                    'end_time': 1e-8,
                    'dt': 1e-8 / 10.5,
                    'print_time': True}
    with pytest.raises(RuntimeError):
        SimulationClock(Simulation({}), clock_config)
```
By using `1e-8 / 10.5` for `dt`, it might help make it clear that the number of steps would not be an integer. 

Thoughts?

Reply:
@arichar6 I think it does make it more clear that it isn't an integer. Thank you!